src/components/Truncate/Truncate.tsx

```diff
@@ -34,16 +35,20 @@ export class Truncate extends React.PureComponent<
   }
 
   componentDidMount() {
-    if (this.props.type === 'auto') {
-      this.setState({
-        isTruncated: this.isTruncated(this.props),
-      })
-    }
+    this._isMounted = true
+    setTimeout(() => {
```

Comment:
Might be good to leave a comment here about why this is being deferred to the next tick.